internal/fs/inode/dir.go

```diff
@@ -94,7 +94,7 @@ type DirInode interface {
 
 	// Create an empty local child file with the supplied (relative) name. Local
 	// file means the object is not yet created in GCS.
-	CreateLocalChildFile(name string) (*Core, error)
+	CreateLocalChildFile(name string, updateInodeFn func(core Core) error) error
```

Comment:
I understand this callback is to make sure, cache insertion happens after creation of local file inode. But I would avoid passing callback, as it will complicate the lock ordering.

Also, I don't see any problem in inserting into the type-cache before inode creation. We were doing similar before only.

@vadlakondaswetha any thought here?